src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -321,6 +321,16 @@ public override Task<bool> HandleRequestAsync()
                 _configuration = await Options.ConfigurationManager.GetConfigurationAsync(Context.RequestAborted);
             }
 
+            if (properties.Items.TryGetValue(OpenIdConnectDefaults.PromptPropertiesKey, out var prompt))
```

Comment:
What other properties should we consider overriding like this? Resource? ResponseType? Scope?

Reply:
This is going to get bulky if we do more than one property. Extract to a helper method?

Reply:
Yeah, we should use this opportunity to look at the other properties. These are the properties that are passed to `OpenIdConnectMessage`:

* `ClientId`, `IssuerAddress`, `ResponseType` – When OIDC clients are registered, these are usually fixed. If there’s a valid reason why one of these should be different for a request, then that should probably be a different auth scheme.
* `RedirectUri` – Is specific to the authentication scheme instance, in order for the scheme to be able to reply it. So it shouldn’t be changeable.
* `ResponseMode` – This affects how the authentication provider responds; so users shouldn’t touch this when they want the OIDC handler to be able to parse the response.
* `Nonce` – This shouldn’t be overwriteable.
* `EnableTelemetryParameters` – Don’t see a valid use case where changing this would be useful.
* `Resource` – I’m actually not sure what this is used for. The spec does not mention such a parameter.
* `Scope`, `MaxAge` – These probably make some sense to overwrite.

Reply:
`Scope` and `MaxAge` do make this a bit complicated though since they are `ICollection<string>` and `TimeSpan?`. So either we need to add some kind of type validation, or expect users to supply it already correctly formatted as strings.

Reply:
Give it a try and we'll see how it looks.